services/ux-backend/handlers/storages/handler.go

```diff
@@ -0,0 +1,143 @@
+	result, err := getStorageInfo(r.Context(), client, namespace)
+	if err != nil {
+		klog.Errorf("failed to get storage info: %v", err)
+		http.Error(w, fmt.Sprintf("Failed to get storage info: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		klog.Errorf("failed to encode response: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+}
+
+type storageInfoResponse struct {
+	OperatorNamespace string                                   `json:"operatorNamespace"`
+	ClusterNamespaces map[string]clusterNamespaceStorageInfo `json:"clusterNamespaces"`
+}
+
+type clusterNamespaceStorageInfo struct {
+	StorageClusterName string `json:"storageClusterName"`
+	DeploymentType     string `json:"deploymentType"`
+	RgWSecureEndpoint  string `json:"rgwSecureEndpoint,omitempty"`
+}
+
+func getStorageInfo(ctx context.Context, client ctrlclient.Client, namespace string) (*storageInfoResponse, error) {
+	klog.Info("Getting storage info")
+
+	storageClusterList := &ocsv1.StorageClusterList{}
+	if err := client.List(ctx, storageClusterList); err != nil {
+		return nil, fmt.Errorf("failed to list StorageClusters: %w", err)
+	}
+
+	cephObjectStoreList := &cephv1.CephObjectStoreList{}
+	if err := client.List(ctx, cephObjectStoreList); err != nil {
+		return nil, fmt.Errorf("failed to list CephObjectStores: %w", err)
+	}
+
+	cosByNamespace := make(map[string]*cephv1.CephObjectStore)
+	for i := range cephObjectStoreList.Items {
+		cos := &cephObjectStoreList.Items[i]
+		cosNamespace := cos.Namespace
+		if _, exists := cosByNamespace[cosNamespace]; !exists {
+			cosByNamespace[cosNamespace] = cos
+		}
+	}
+
+	result := &storageInfoResponse{
+		OperatorNamespace: namespace,
+		ClusterNamespaces: make(map[string]clusterNamespaceStorageInfo),
+	}
+
+	for _, sc := range storageClusterList.Items {
+		scNamespace := sc.Namespace
+		namespaceInfo := clusterNamespaceStorageInfo{
+			StorageClusterName: sc.Name,
+			DeploymentType:     getDeploymentType(&sc),
+		}
+
+		if cos, exists := cosByNamespace[scNamespace]; exists {
+			rgwEndpoint := getRgWSecureEndpoint(cos)
+			if rgwEndpoint != "" {
+				namespaceInfo.RgWSecureEndpoint = rgwEndpoint
+			}
+		}
+
+		result.ClusterNamespaces[scNamespace] = namespaceInfo
+	}
+
+	return result, nil
+}
+
+func getDeploymentType(sc *ocsv1.StorageCluster) string {
+	if sc.Spec.ExternalStorage.Enable {
+		return deploymentTypeExternal
+	}
+
+	return deploymentTypeInternal
+}
+
+func getRgWSecureEndpoint(cos *cephv1.CephObjectStore) string {
+	if cos.Status == nil {
+		return ""
+	}
+
+	if cos.Status.Info != nil {
+		if endpoint, exists := cos.Status.Info["secureEndpoint"]; exists && endpoint != "" {
+			return endpoint
+		}
+	}
+
+	if len(cos.Status.Endpoints.Secure) > 0 {
+		return cos.Status.Endpoints.Secure[0]
+	}
```

Comment:
```suggestion
	if cos.Status == nil {
		return ""
	}

	if endpoint := cos.Status.Info["secureEndpoint"]; endpoint != "" {
		return endpoint
	}

	if len(cos.Status.Endpoints.Secure) > 0 {
		return cos.Status.Endpoints.Secure[0]
	}
```
Will there be different values at cos.Status.Endpoints.Secure and cos.Status.Info["secureEndpoint"]?

Reply:
another example of nit-pickyness :) but I will update it because it's cleaner...

> Will there be different values at cos.Status.Endpoints.Secure and cos.Status.Info["secureEndpoint"]

No, it's just a fallback.

Reply:
Can there be a case where we can have one field populated and the other one isn't?

Reply:
It would be good to know if there is a case, if there can be a scenario like this, and if yes, then which endpoint should we give a priority to? If it cannot be different, then let's use one of the fields. Why write more code than what is required?